### PR TITLE
Add contracts to bridge draw

### PR DESCRIPTION
**Pull request by PierrickGT**
Architecture: https://www.figma.com/file/IE4GGo8q6XltiC58HL24Ij/V4-1.2.0-Bridge-Draw-Architecture?node-id=0%3A1&t=DizRA8YRx6rdXtDt-1

Add `DrawRelayer` and `DrawExecutor` contracts that relies on ERC-5164 to bridge draws from Ethereum to any L1 or L2 where Chainlink VRF 2.0 wouldn't be available.
`DrawExecutor` does not ensure draw ordering and draws should always be bridged in ascending and contiguous order.

### CODE REVIEW

**Review by asselstine — COMMENTED** [review]
Is it not possible to reuse the contracts in the ERC5164 repo?

It LGTM but we'll prob want to wait for the audit to complete

**Comment by PierrickGT** [comment]
> Is it not possible to reuse the contracts in the ERC5164 repo?
> 
> It LGTM but we'll prob want to wait for the audit to complete

It's possible once we release a package for ERC-5164.
I'll update this PR with the package after completing the audit fixes.

File: contracts/abstract/ExecutorAware.sol [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity 0.8.6;
+
+/**
+ * @title ExecutorAware abstract contract
+ * @notice The ExecutorAware contract allows contracts on a receiving chain to execute calls from an origin chain.
+ *         These calls are sent by the `CrossChainRelayer` contract which live on the origin chain.
+ *         The `CrossChainExecutor` contract on the receiving chain executes these calls
+ *         and then forward them to an ExecutorAware contract on the receiving chain.
```
Comment (dylandesrosier) [inline contracts/abstract/ExecutorAware.sol:10]:
```suggestion
 *         and then forwards them to an ExecutorAware contract on the receiving chain.
```

Comment (PierrickGT) [reply contracts/abstract/ExecutorAware.sol]: Done: https://github.com/pooltogether/v4-periphery/pull/61/commits/243ecc39dc6349e9e26ae6c3716a2025f08e3a34

**Comment by PierrickGT** [comment]
> Is it not possible to reuse the contracts in the ERC5164 repo?
Well actually no, it is not possible since we use Solidity 0.8.16 in the ERC-5164 repo and 0.8.6 in this one.

**Pull request merged**